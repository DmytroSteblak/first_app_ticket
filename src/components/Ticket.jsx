import React from 'react';
import { Card, Col, Image, Row } from 'react-bootstrap';

import { CardBody } from '../styled/Ticket-styled';

const Ticket = ({
    price,
    duration,
    durationBack,
    date,
    dateBack,
    stopsAmount,
    stopsBackAmount,
    origin,
    originBack,
    destination,
    destinationBack,
    stops,
    stopsBack
}) => {
    const getTime = (duration) => {
        return Math.floor(duration / 60) + 'ч: ' + (duration % 60) + 'м';
    };
    const getData = (date, duration) => {
        const dateStart = new Date(date);
        const dateEnd = new Date(dateStart.getTime() + duration * 60 * 1000);
        return dateStart.toTimeString().slice(0, 5) + ' - ' + dateEnd.toTimeString().slice(0, 5);
    };

    return (
        <>
            <CardBody>
                <Card className="mt-4 border-0 shadow 1 ">
                    <Card.Body>
                        <Row className="header_ticket m-1">
                            <h5 className="w-50">{price}p</h5>
                            <Image src="/assets/icon.png"/>
                        </Row>
                        <Row className="content_ticket">
                            <Col lg="3" md="3" sm="4" xs="5" className="m-2">
                                <div className="topClass">
                                    {origin} - {destination}
                                </div>
                                <div className="hideClass">
                                    {getData(date, duration)}
                                </div>
                            </Col>
                            <Col lg="3" md="3" sm="4" xs="5" className="m-2">
                                <div className="topClass">
                                    в пути
                                </div>
                                <div className="hideClass">
                                    {getTime(duration)}
                                </div>
                            </Col>
                            <Col lg="3" md="3" sm="4" xs="5" className="m-2">
                                <div className="topClass">
                                    {stopsAmount} пересадки
                                </div>
                                <div className="hideClass">
                                    {stops}
                                </div>
                            </Col>
                            <Col lg="3" md="3" sm="4" xs="5" className="m-2">
                                <div className="topClass">
                                    {originBack} - {destinationBack}
                                </div>
                                <div className="hideClass">
                                    {getData(dateBack, durationBack)}
                                </div>
                            </Col>
                            <Col lg="3" md="3" sm="4" xs="5" className="m-2">
                                <div className="topClass">
                                    в пути
                                </div>
                                <div className="hideClass">
                                    {getTime(durationBack)}
                                </div>
                            </Col>
                            <Col lg="3" md="3" sm="4" xs="5" className="m-2">
                                <div className="topClass">
                                    {stopsBackAmount} пересадки
                                </div>
                                <div className="hideClass">
                                    {stopsBack}
                                </div>
                            </Col>
                        </Row>
                    </Card.Body>
                </Card>
            </CardBody>
        </>
    );
};

export default Ticket;