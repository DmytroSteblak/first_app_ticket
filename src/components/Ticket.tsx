import React from 'react';
import { Card, Col, Image, Row } from 'react-bootstrap';

import { getData, getTime } from '../different';
import { PropsTickets } from '../types';
import { CardBody } from '../styled/Ticket-styled';

const Ticket: React.FC<PropsTickets> = ({
    price,
    duration,
    durationBack,
    date,
    dateBack,
    stopsAmount,
    stopsBackAmount,
    origin,
    originBack,
    destination,
    destinationBack,
    stops,
    stopsBack
}) => {

    return (
        <>
            <CardBody>
                <Card className="mt-4 border-0 shadow 1 ">
                    <Card.Body>
                        <Row className="header_ticket m-1">
                            <h5 className="w-50">{price}p</h5>
                            <Image src="/assets/icon.png"/>
                        </Row>
                        <Row className="content_ticket">
                            <Col lg="3" md="3" sm="4" xs="5" className="m-2">
                                <div className="topClass">
                                    {origin} - {destination}
                                </div>
                                <div className="hideClass">
                                    {getData(date, duration)}
                                </div>
                            </Col>
                            <Col lg="3" md="3" sm="4" xs="5" className="m-2">
                                <div className="topClass">
                                    в пути
                                </div>
                                <div className="hideClass">
                                    {getTime(duration)}
                                </div>
                            </Col>
                            <Col lg="3" md="3" sm="4" xs="5" className="m-2">
                                <div className="topClass">
                                    {stopsAmount} пересадки
                                </div>
                                <div className="hideClass">
                                    {stops}
                                </div>
                            </Col>
                            <Col lg="3" md="3" sm="4" xs="5" className="m-2">
                                <div className="topClass">
                                    {originBack} - {destinationBack}
                                </div>
                                <div className="hideClass">
                                    {getData(dateBack, durationBack)}
                                </div>
                            </Col>
                            <Col lg="3" md="3" sm="4" xs="5" className="m-2">
                                <div className="topClass">
                                    в пути
                                </div>
                                <div className="hideClass">
                                    {getTime(durationBack)}
                                </div>
                            </Col>
                            <Col lg="3" md="3" sm="4" xs="5" className="m-2">
                                <div className="topClass">
                                    {stopsBackAmount} пересадки
                                </div>
                                <div className="hideClass">
                                    {stopsBack}
                                </div>
                            </Col>
                        </Row>
                    </Card.Body>
                </Card>
            </CardBody>
        </>
    );
};

export default Ticket;
