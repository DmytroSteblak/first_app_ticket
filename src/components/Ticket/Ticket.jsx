import React from "react";
import {Card, Col, Image, Row} from "react-bootstrap";
import {CardBody} from "../../styled/Ticket-styled";



const Ticket = () =>(
    <CardBody>
        <Card className="mt-4 border-0 shadow 1 ">
            <Card.Body>
                <Row className="header_ticket m-1">
                    <h5 className="w-50">12 000p</h5>
                    <Image src="/assets/icon.png"/>
                </Row>

                <Row className="content_ticket">
                    <Col lg="3" md="3" sm="4" xs="5" className="m-2">
                        <div className="topClass">
                            mow - hkt
                        </div>
                        <div className="hideClass">
                            10:45-08:00
                        </div>
                    </Col>
                    <Col lg="3" md="3" sm="4" xs="5" className="m-2">
                        <div className="topClass">
                            в пути
                        </div>
                        <div className="hideClass">
                            21ч 15м
                        </div>
                    </Col>
                    <Col lg="3" md="3" sm="4" xs="5" className="m-2">
                        <div className="topClass">
                            2 пересадки
                        </div>
                        <div className="hideClass">
                            HKG,JNB
                        </div>
                    </Col>
                    <Col lg="3" md="3" sm="4" xs="5" className="m-2">
                        <div className="topClass">
                            mow - hkt
                        </div>
                        <div className="hideClass">
                            11:20-00:50
                        </div>
                    </Col>
                    <Col lg="3" md="3" sm="4" xs="5" className="m-2">
                        <div className="topClass">
                            в пути
                        </div>
                        <div className="hideClass">
                            13ч 30м
                        </div>
                    </Col>
                    <Col lg="3" md="3" sm="4" xs="5" className="m-2">
                        <div className="topClass">
                            2 пересадки
                        </div>
                        <div className="hideClass">
                            HKG
                        </div>
                    </Col>
                </Row>
                {/*<Row className="d-flex justify-content-between">*/}

                {/*</Row>*/}
            </Card.Body>
        </Card>
    </CardBody>



)

export default Ticket;