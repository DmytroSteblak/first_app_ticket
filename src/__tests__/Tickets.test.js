import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';

import Ticket from '../components/Ticket';

const props = {
    price: '15400',
    duration: 1889,
    durationBack: 738,
    date: '2021-08-14T08:42:00.000Z',
    dateBack: '2021-09-03T06:59:00.000Z',
    stopsAmount: 2,
    stopsBackAmount: 3,
    origin: 'MOW',
    originBack: 'HKT',
    destination: 'HTK',
    destinationBack: 'MOW',
    stops: 'BKK',
    stopsBack: 'DBX,KUL'
};

describe('Testing component <Ticket />', () => {

    it('should render Ticket component with props', function () {
        const component = render(<Ticket {...props} />);
        expect(toJson(component)).toMatchSnapshot();
    });

    it('should render Ticket component without props', function () {
        const component = render(<Ticket />);
        expect(toJson(component)).toMatchSnapshot();
    });

});
