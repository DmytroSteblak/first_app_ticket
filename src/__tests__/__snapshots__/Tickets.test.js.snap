// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component <Ticket /> should render Ticket component with props 1`] = `
<div
  class="sc-gtsrHT htsKDc"
>
  <div
    class="mt-4 border-0 shadow 1  card"
  >
    <div
      class="card-body"
    >
      <div
        class="header_ticket m-1 row"
      >
        <h5
          class="w-50"
        >
          15400p
        </h5>
        <img
          class=""
          src="/assets/icon.png"
        />
      </div>
      <div
        class="content_ticket row"
      >
        <div
          class="m-2 col-lg-3 col-md-3 col-sm-4 col-5"
        >
          <div
            class="topClass"
          >
            MOW - HTK
          </div>
          <div
            class="hideClass"
          >
            11:42 - 19:11
          </div>
        </div>
        <div
          class="m-2 col-lg-3 col-md-3 col-sm-4 col-5"
        >
          <div
            class="topClass"
          >
            в пути
          </div>
          <div
            class="hideClass"
          >
            31ч: 29м
          </div>
        </div>
        <div
          class="m-2 col-lg-3 col-md-3 col-sm-4 col-5"
        >
          <div
            class="topClass"
          >
            2 пересадки
          </div>
          <div
            class="hideClass"
          >
            BKK
          </div>
        </div>
        <div
          class="m-2 col-lg-3 col-md-3 col-sm-4 col-5"
        >
          <div
            class="topClass"
          >
            HKT - MOW
          </div>
          <div
            class="hideClass"
          >
            09:59 - 22:17
          </div>
        </div>
        <div
          class="m-2 col-lg-3 col-md-3 col-sm-4 col-5"
        >
          <div
            class="topClass"
          >
            в пути
          </div>
          <div
            class="hideClass"
          >
            12ч: 18м
          </div>
        </div>
        <div
          class="m-2 col-lg-3 col-md-3 col-sm-4 col-5"
        >
          <div
            class="topClass"
          >
            3 пересадки
          </div>
          <div
            class="hideClass"
          >
            DBX,KUL
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing component <Ticket /> should render Ticket component without props 1`] = `
<div
  class="sc-gtsrHT htsKDc"
>
  <div
    class="mt-4 border-0 shadow 1  card"
  >
    <div
      class="card-body"
    >
      <div
        class="header_ticket m-1 row"
      >
        <h5
          class="w-50"
        >
          p
        </h5>
        <img
          class=""
          src="/assets/icon.png"
        />
      </div>
      <div
        class="content_ticket row"
      >
        <div
          class="m-2 col-lg-3 col-md-3 col-sm-4 col-5"
        >
          <div
            class="topClass"
          >
             - 
          </div>
          <div
            class="hideClass"
          >
            Inval - Inval
          </div>
        </div>
        <div
          class="m-2 col-lg-3 col-md-3 col-sm-4 col-5"
        >
          <div
            class="topClass"
          >
            в пути
          </div>
          <div
            class="hideClass"
          >
            NaNч: NaNм
          </div>
        </div>
        <div
          class="m-2 col-lg-3 col-md-3 col-sm-4 col-5"
        >
          <div
            class="topClass"
          >
             пересадки
          </div>
          <div
            class="hideClass"
          />
        </div>
        <div
          class="m-2 col-lg-3 col-md-3 col-sm-4 col-5"
        >
          <div
            class="topClass"
          >
             - 
          </div>
          <div
            class="hideClass"
          >
            Inval - Inval
          </div>
        </div>
        <div
          class="m-2 col-lg-3 col-md-3 col-sm-4 col-5"
        >
          <div
            class="topClass"
          >
            в пути
          </div>
          <div
            class="hideClass"
          >
            NaNч: NaNм
          </div>
        </div>
        <div
          class="m-2 col-lg-3 col-md-3 col-sm-4 col-5"
        >
          <div
            class="topClass"
          >
             пересадки
          </div>
          <div
            class="hideClass"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
